packages/loader/container-loader/src/container.ts

```diff
@@ -2128,10 +2130,13 @@ export class Container
 				checkpointSequenceNumber,
 				quorumSize: this._protocolHandler?.quorum.getMembers().size,
 				isDirty: this.isDirty,
+				noopCount: this.noopCount,
 				...this._deltaManager.connectionProps,
 			},
 			reason?.error,
 		);
+		// Reset noop count after it has been logged
+		this.noopCount = 0;
```

Comment:
I'd probably reset it only on disconnect.
I think we would send noops in ConnectionState.CatchingUp state and it would be great to keep counting (and have good count) at disconnect event for whole connection.
That said, this likely means we want to have this property only on connected events. There is no harm to report it on CatchingUp event, but someone somewhere will make a mistake and will count it twice.

The way you have it is also fine, but one needs to remember that total number of noops comes from a sum of properties on both CatchingUp & Disconnect events.

Reply:
Why are we sending this only on connection? Clients that don't disconnect / reconnect will not send this telemetry right. Is that okay?

Reply:
You are asking valid question, but unrelated to my comment. My comment - if we send it on connection change events, do not reset counter on "CatchingUp" transition. As for core of your comment - see next comment where one of the possible directions we could make is do sampling at noop sending.